beetsplug/spotify.py

```diff
@@ -404,34 +409,31 @@ def _construct_search_query(filters=None, keywords=""):
             query = query.decode("utf8")
         return unidecode.unidecode(query)
 
-    def _search_api(self, query_type, filters=None, keywords=""):
+    def _search_api(
+        self,
+        query_type: Literal["album", "track"],
+        filters: dict[str, str],
+        keywords="",
+    ) -> Sequence[Response]:
         """Query the Spotify Search API for the specified ``keywords``,
         applying the provided ``filters``.
 
         :param query_type: Item type to search across. Valid types are:
             'album', 'artist', 'playlist', and 'track'.
-        :type query_type: str
         :param filters: (Optional) Field filters to apply.
-        :type filters: dict
         :param keywords: (Optional) Query keywords to use.
-        :type keywords: str
-        :return: JSON data for the class:`Response <Response>` object or None
-            if no search results are returned.
-        :rtype: dict or None
         """
         query = self._construct_search_query(keywords=keywords, filters=filters)
-        if not query:
-            return None
```

Comment:
Maybe this should be replaced by a `if not keywords` check before the call to `_construct_search_query` instead of removing the check entirely?

Reply:
It is fine if no keywords are given afaiks.  The `_construct_search_query` method  only requires filters, keywords are optional.

Reply:
My concern was that (at least superficially), it seems like the code previously prevented a completely empty query being sent to the server (which I'd guess might have unintended side-effects, if only by yielding completely random data).
Maybe there's no way such a query would ever be constructed by our code and that's fine?

In any case, I don't know the code around metadata handling sufficiently well to say much about this; I'll leave this up to your judgement :)

Reply:
I think this None check was some historical code that can never be reached in the current version. A query is always returned by `_construct_search_query` as we always give a filter. At least in the beets first part code. Maybe it was different at some point but at the moment it shouldn't be an issue as far as I can see.

In future if we run into issues regarding this, I would opt to just raise an error in the `_construct_search_query` function instead. Will keep my eyes open for issues regarding this :+1: